packages/x-data-grid/src/components/toolbar/GridToolbarExport.tsx

```diff
@@ -32,7 +32,10 @@ export interface GridToolbarExportProps {
    * @default {}
    */
   slotProps?: { button?: Partial<ButtonProps>; tooltip?: Partial<TooltipProps> };
-  [key: string]: any;
+  // TODO v8: Remove this loophole
+  // Refactored from: [key: string]: any;
+  [x: `data-${string}`]: string;
+  [x: `${string}Options`]: any;
```

Comment:
WDYT about this approach?
Can we keep just the second line?
Or only `excelOptions`? 🤔